packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -0,0 +1,339 @@
+      collateralSold = settle(seat, want);
+
+      if (AmountMath.isEmpty(seat.getCurrentAllocation().Currency)) {
+        seat.exit();
+        return;
+      }
+    }
+
+    const stillWant = AmountMath.subtract(want, collateralSold);
+    if (!AmountMath.isEmpty(stillWant)) {
+      trace('added Offer ', price, stillWant.value);
+      priceBook.add(seat, price, stillWant);
+    } else {
+      seat.exit();
+    }
+  };
+
+  /**
+   *  Accept an offer expressed as a discount (or markup). If the auction is
+   *  active, attempt to buy collateral. If any of the offer remains add it to
+   *  the book.
+   *
+   *  @param {ZCFSeat} seat
+   *  @param {Ratio} bidScaling
+   *  @param {Amount} want
+   *  @param {AuctionState} auctionState
+   */
+  const acceptScaledBidOffer = (seat, bidScaling, want, auctionState) => {
+    trace('accept scaled bid offer');
+    let collateralSold = AmountMath.makeEmptyFromAmount(want);
+
+    if (
+      isActive(auctionState) &&
+      isScaledBidPriceHigher(bidScaling, curAuctionPrice, lockedPriceForRound)
+    ) {
+      collateralSold = settle(seat, want);
+      if (AmountMath.isEmpty(seat.getCurrentAllocation().Currency)) {
+        seat.exit();
+        return;
+      }
+    }
+
+    const stillWant = AmountMath.subtract(want, collateralSold);
+    if (!AmountMath.isEmpty(stillWant)) {
+      scaledBidBook.add(seat, bidScaling, stillWant);
+    } else {
+      seat.exit();
+    }
+  };
+
+  return Far('AuctionBook', {
+    addAssets(assetAmount, sourceSeat) {
+      trace('add assets');
+      assetsForSale = AmountMath.add(assetsForSale, assetAmount);
+      atomicRearrange(
+        zcf,
+        harden([[sourceSeat, collateralSeat, { Collateral: assetAmount }]]),
+      );
+    },
+    settleAtNewRate(reduction) {
+      curAuctionPrice = multiplyRatios(reduction, lockedPriceForRound);
+
+      const pricedOffers = priceBook.offersAbove(curAuctionPrice);
+      const discOffers = scaledBidBook.offersAbove(reduction);
+
+      // requested price or bid scaling gives no priority beyond specifying which
+      // round the order will be service in.
+      const prioritizedOffers = [...pricedOffers, ...discOffers].sort();
+
+      trace(`settling`, pricedOffers.length, discOffers.length);
+      for (const [key, { seat, price: p, wanted }] of prioritizedOffers) {
+        if (seat.hasExited()) {
+          removeFromOneBook(p, key);
+        } else {
+          const collateralSold = settle(seat, wanted);
+
+          if (
+            AmountMath.isEmpty(seat.getCurrentAllocation().Currency) ||
+            AmountMath.isGTE(seat.getCurrentAllocation().Collateral, wanted)
+          ) {
+            seat.exit();
+            removeFromOneBook(p, key);
+          } else if (!AmountMath.isGTE(collateralSold, wanted)) {
+            if (p) {
+              priceBook.updateReceived(key, collateralSold);
+            } else {
+              scaledBidBook.updateReceived(key, collateralSold);
+            }
```

Comment:
for parity with "remove",
```suggestion
            updateItsBook(key, collateralSold, p);
```

Reply:
done